src/main/java/seedu/toluist/controller/HistoryController.java

```diff
@@ -0,0 +1,48 @@
+package seedu.toluist.controller;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+import seedu.toluist.commons.core.LogsCenter;
+import seedu.toluist.dispatcher.CommandResult;
+import seedu.toluist.ui.Ui;
+
+/**
+ * ListController is responsible for rendering the initial UI
+ */
+public class HistoryController extends Controller {
+    private static final Logger logger = LogsCenter.getLogger(ClearController.class);
+    private static final String RESULT_MESSAGE = "%s\n%d commands displayed.";
+    private static final String COMMAND_WORD = "history";
+    private static final String COMMAND_REGEX = "^history\\s*";
+
+    public HistoryController(Ui renderer) {
+        super(renderer);
+    }
+
+    @Override
+    public CommandResult execute(String command) {
+        return null; //not used
+    }
+
+    public CommandResult execute(String command, ArrayList<String> commandHistory) {
+        logger.info(getClass().getName() + " will handle command");
```

Comment:
rather than having a separate execute method with differen signature, a more conformant way is to have a `setCommandHistory`, using setter injection to pass the command history over. Then `execute` api is not a dummy method anymore